services/business/health.go

```diff
@@ -15,27 +16,52 @@ type HealthService struct {
 }
 
 // GetServiceHealth returns a service health from just namespace and service (thus, it fetches data from K8S and Prometheus)
-func (in *HealthService) GetServiceHealth(namespace, service string) models.Health {
+func (in *HealthService) GetServiceHealth(namespace, service, rateInterval string) models.Health {
+	// Fill all parts
 	health := models.Health{}
+	in.fillMissingParts(namespace, service, rateInterval, &health)
+	return health
+}
 
-	// Pod status
-	details, _ := in.k8s.GetServiceDetails(namespace, service)
-	if details != nil {
-		health.DeploymentStatuses = castDeploymentsStatuses(&details.Deployments.Items)
-	}
+func (in *HealthService) fillMissingParts(namespace, service, rateInterval string, health *models.Health) {
+	// Pod statuses
+	health.FillDeploymentStatusesIfMissing(func() []models.DeploymentStatus {
```

Comment:
These supplier are a little bit odd, or in other words, pretty functional.
Is there anything else beyond just to assign the status or is t just style ?

Reply:
It's a way to prevent the leak of the "Fetched" flag & the associated logic (check & set boolean) outside of the model.